app/src/main/java/com/andrew/tasky/presentation/adapter/AgendaItemAdapter.kt

```diff
@@ -0,0 +1,62 @@
+package com.andrew.tasky.presentation.adapter
+
+import android.graphics.Color
+import android.graphics.Paint
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.andrew.tasky.R
+import com.andrew.tasky.data.AgendaItem
+import com.andrew.tasky.databinding.ItemAgendaBinding
+import com.andrew.tasky.util.AgendaItemType
+
+class AgendaItemAdapter(
+    var agendaItems: List<AgendaItem>
+): RecyclerView.Adapter<AgendaItemAdapter.AgendaItemViewHolder>() {
+
+    inner class AgendaItemViewHolder(val binding: ItemAgendaBinding) : RecyclerView.ViewHolder(binding.root)
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AgendaItemViewHolder {
+        val layoutInflater = LayoutInflater.from(parent.context)
+        val binding = ItemAgendaBinding.inflate(layoutInflater, parent, false)
+        return AgendaItemViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: AgendaItemViewHolder, position: Int) {
+        holder.binding.apply {
+            agendaItemTitle.text = agendaItems[position].title
+            agendaItemDescription.text = agendaItems[position].description
+            agendaItemDate.text = agendaItems[position].fromDate + ", " + agendaItems[position].fromTime
```

Comment:
In general, I always keep times and dates as `LocalDateTime` in my models. That has the advantage that it's up to the UI how to display it. You might want to have different format patterns on different screens which doesn't work if you format once and attach the date as String to the model :)